code/modules/mob/living/basic/pets/dog.dm

```diff
@@ -619,45 +633,76 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
 	fdel(json_file)
 	WRITE_FILE(json_file, json_encode(file_data))
 
-/mob/living/basic/pet/dog/corgi/ian/narsie_act()
-	playsound(src, 'sound/magic/demon_dies.ogg', 75, TRUE)
-	var/mob/living/basic/pet/dog/corgi/narsie/N = new(loc)
-	N.setDir(dir)
-	investigate_log("has been gibbed by Nar'Sie.", INVESTIGATE_DEATHS)
-	gib()
+///Reads the database's persistence json file and applies age, record_age and saved_head to Ian.
+/mob/living/basic/pet/dog/corgi/ian/proc/Read_Memory()
+	if(fexists("data/npc_saves/Ian.sav")) //legacy compatability to convert old format to new
+		var/savefile/S = new /savefile("data/npc_saves/Ian.sav")
+		S["age"] >> age
+		S["record_age"] >> record_age
+		S["saved_head"] >> saved_head
+		fdel("data/npc_saves/Ian.sav")
+	else
+		var/json_file = file("data/npc_saves/Ian.json")
+		if(!fexists(json_file))
+			return
+		var/list/json = json_decode(file2text(json_file))
+		age = json["age"]
+		record_age = json["record_age"]
+		saved_head = json["saved_head"]
+	if(isnull(age))
+		age = 0
+	if(isnull(record_age))
+		record_age = 1
+	if(saved_head)
+		place_on_head(new saved_head)
 
+///Checks whether Ian has survived the round or not
+/mob/living/basic/pet/dog/corgi/ian/proc/check_Ian_survival()
+	if(!stat && !memory_saved)
+		Write_Memory(FALSE)
+
+//NARS-IAN! SQ-Q-QooEglor-r'EEn-nl-luEEEf-f-fth-h
 /mob/living/basic/pet/dog/corgi/narsie
 	name = "Nars-Ian"
+	real_name = "Nars-Ian"
 	desc = "Ia! Ia!"
 	icon_state = "narsian"
 	icon_living = "narsian"
 	icon_dead = "narsian_dead"
 	faction = list(FACTION_NEUTRAL, FACTION_CULT)
+	speak_emote = list("growls", "barks ominously")
 	gold_core_spawnable = NO_SPAWN
-	nofur = TRUE
+	can_be_shaved = FALSE
 	unique_pet = TRUE
 	held_state = "narsian"
 
+//this could maybe be turned into an element for sanity but i don't want to get too distracted
+//i mean i was just placing butter on a corgi yet here i am now
```

Comment:
```suggestion
```

sorry but code comments should be pertinent. just leave a "TODO make element" or something